src/main/GroceryList.java

```diff
@@ -58,9 +58,9 @@ public GroceryItem getItem(String itemName) {
 		return null;
 	}
 
-	public void editItem(GroceryItem updatedItem, GroceryItem oldItem) {
-		int indexToReplace = this.items.indexOf(oldItem);
-		this.items.set(indexToReplace, updatedItem);
+	public boolean editItem(GroceryItem updatedItem, GroceryItem oldItem) {
+		this.items.remove(oldItem);

```

Comment:
Make sure you remove the costs of the old items in addition to remove the item from the set